src/adam_op/adam_op_impl.cpp

```diff
@@ -31,7 +31,7 @@ void adamForwardInplaceCPUKernel(
     const other_t inv_one_minus_pow_b2, const other_t eps,
     const other_t eps_root, const size_t n, scalar_t* __restrict__ updates_ptr,
     scalar_t* __restrict__ mu_ptr, scalar_t* __restrict__ nu_ptr) {
-#pragma omp parallel for num_threads(32)
+#pragma omp parallel for num_threads(omp_get_num_procs())
```

Comment:
Is this machine dependent? We are building pypi wheels in GitHub Action containers.

Reply:
no, depending on the user's machine (runtime decided)